pkg/geo/geofn.go

```diff
@@ -0,0 +1,37 @@
+// Copyright 2020 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package geo
+
+import (
+	"github.com/cockroachdb/errors"
+	"github.com/golang/geo/s2"
+	"github.com/twpayne/go-geom"
+)
+
+// STAreaGeometry exactly matches the semantics of ST_Area.
+func STAreaGeometry(g *Geometry) (float64, error) {
+	switch s := g.Shape.(type) {
+	case *geom.Point:
+		return s.Area(), nil
+	default:
+		return 0.0, errors.Errorf(`unhandled geom type %T: %s`, s, s)
+	}
+}
+
+// STAreaGeography exactly matches the semantics of ST_Area.
+func STAreaGeography(g *Geography) (float64, error) {
```

Comment:
i'm thinking we should make these methods on `Geography` / `Geometry` instead of their separate packages, and I know we've discussed this before. what made you decide to end up here?

Reply:
This is the same package (for now) but functions instead of methods. My thought is that pkg/geo should be a general-ish purpose lib. This means common calculations (like area, etc) should live here, but the more obscure things we'd need only for SQL should eventually be moved to a separate package.

Reply:
i'm thinking we probably want to separate it for `geogfn` and `geomfn`, if we go down this route.